hoomd/md/compute.py

```diff
@@ -0,0 +1,276 @@
+
+# Maintainer: joaander / All Developers are free to add commands for new features
+
+r""" Compute system properties
+
+A compute calculates properties of the system on demand. Most computes are automatically created by the object
+that needs them (e.g. methods.NVT creates a compute.ThermodynamicQuantities for temperature calculations). User-specified
+computes can be used when more flexibility is needed. Properties calculated by specified computes (automatically, or by the
+user) can be logged using hoomd.logging.Logger.
+"""
+
+from hoomd import _hoomd;
+from hoomd.md import _md
+from hoomd.operation import _Compute
+from hoomd.logging import log
+import hoomd;
+import sys;
+
+class _Thermo(_Compute):
+
+    def __init__(self, filter):
+        self._filter = filter
+
+
+class ThermodynamicQuantities(_Thermo):
+    R""" Compute thermodynamic properties of a group of particles.
+
+    Args:
+        filter (``hoomd.filter``): Particle filter to compute thermodynamic properties for.
+
+    :py:class:`ThermodynamicQuantities` acts on a given group of particles and calculates thermodynamic properties of
+    those particles when requested. All specified :py:class:`ThermodynamicQuantities` objects can be added to a logger
+    for logging during a simulation, see :py:class:`hoomd.logging.Logger` for more details.
+
+    Examples::
+
+        f = filter.Type('A')
+        compute.ThermodynamicQuantities(filter=f)
+    """
+
+    def __init__(self, filter):
+        super().__init__(filter)
+
+    def _attach(self):
+        if isinstance(self._simulation.device, hoomd.device.CPU):
+            thermo_cls = _md.ComputeThermo
+        else:
+            thermo_cls = _md.ComputeThermoGPU
+        group = self._simulation.state.get_group(self._filter)
+        self._cpp_obj = thermo_cls(self._simulation.state._cpp_sys_def, group, "")
+        super()._attach()
+
+    @log
+    def kinetic_temperature(self):
+        R"""
+        :math:`kT`, instantaneous thermal energy of the group (in energy units).
+
+        Calculated as
+
+          .. math::
+
+            kT = 2 \cdot \frac{K}{N_{\mathrm{dof}}}
+        """
+        if self._attached:
+            self._cpp_obj.compute(self._simulation.timestep)
+            return self._cpp_obj.kinetic_temperature
+        else:
+            return None
+
+    @log
+    def pressure(self):
+        R"""
+        :math:`P`, instantaneous pressure of the group (in pressure units).
+
+        Calculated as
+
+          .. math::
+
+              W = \frac{1}{2} \sum_{i}\sum_{j \ne i} \vec{F}_{ij} \cdot \vec{r_{ij}} + \sum_{k} \vec{F}_{k} \cdot \vec{r_{k}}
+
+          where :math:`\vec{F}_{ij}` are pairwise forces between particles and :math:`\vec{F}_k` are forces due to explicit constraints, implicit rigid
+          body constraints, external walls, and fields. In 2D simulations,
+          :math:`P = (K + \frac{1}{2}\cdot W)/A` where :math:`A` is the area of the simulation box.
```

Comment:
Also list the `P` expression for 3D boxes. Or better yet, list one general expression that applies to both 2D and 3D.